src/server/game/Misc/BanManager.cpp

```diff
@@ -0,0 +1,316 @@
+        return BAN_LONGER_EXISTS;
+
+    // make sure there is only one active ban
+    PreparedStatement * stmt = LoginDatabase.GetPreparedStatement(LOGIN_UPD_ACCOUNT_NOT_BANNED);
+    stmt->setUInt32(0, AccountID);
+    trans->Append(stmt);
+
+    // No SQL injection with prepared statements
+    stmt = LoginDatabase.GetPreparedStatement(LOGIN_INS_ACCOUNT_BANNED);
+    stmt->setUInt32(0, AccountID);
+    stmt->setUInt32(1, DurationSecs);
+    stmt->setString(2, Author);
+    stmt->setString(3, Reason);
+    trans->Append(stmt);
+
+    if (WorldSession* session = sWorld->FindSession(AccountID))
+        if (session->GetPlayerName() != Author)
+            session->KickPlayer("Ban Account at condition 'FindSession(account)->GetPlayerName() != author'");
+
+    if (WorldSession* session = sWorld->FindOfflineSession(AccountID))
+        if (session->GetPlayerName() != Author)
+            session->KickPlayer("Ban Account at condition 'FindOfflineSession(account)->GetPlayerName() != author'");
+
+    LoginDatabase.CommitTransaction(trans);
+
+    if (sWorld->getBoolConfig(CONFIG_SHOW_BAN_IN_WORLD))
+    {
+        bool IsPermanetly = true;
+
+        if (TimeStringToSecs(Duration) > 0)
+            IsPermanetly = false;
+
+        std::string AccountName;
+
+        AccountMgr::GetName(AccountID, AccountName);
+
+        if (!IsPermanetly)
+            sWorld->SendWorldText(LANG_BAN_ACCOUNT_YOUBANNEDMESSAGE_WORLD, Author.c_str(), AccountName.c_str(), secsToTimeString(TimeStringToSecs(Duration), true).c_str(), Reason.c_str());
+        else
+            sWorld->SendWorldText(LANG_BAN_ACCOUNT_YOUPERMBANNEDMESSAGE_WORLD, Author.c_str(), AccountName.c_str(), Reason.c_str());
+    }
+
+    return BAN_SUCCESS;
+}
+
+/// Ban an IP address, duration will be parsed using TimeStringToSecs if it is positive, otherwise permban
+BanReturn BanManager::BanIP(std::string const& IP, std::string const& Duration, std::string const& Reason, std::string const& Author)
+{
+    if (IP.empty() || Duration.empty())
+        return BAN_SYNTAX_ERROR;
+
+    uint32 DurationSecs = TimeStringToSecs(Duration);
+
+    // No SQL injection with prepared statements
+    PreparedStatement* stmt = LoginDatabase.GetPreparedStatement(LOGIN_SEL_ACCOUNT_BY_IP);
+    stmt->setString(0, IP);
+    PreparedQueryResult resultAccounts = LoginDatabase.Query(stmt);
+
+    stmt = LoginDatabase.GetPreparedStatement(LOGIN_INS_IP_BANNED);
+    stmt->setString(0, IP);
+    stmt->setUInt32(1, DurationSecs);
+    stmt->setString(2, Author);
+    stmt->setString(3, Reason);
+    LoginDatabase.Execute(stmt);
+
+    if (sWorld->getBoolConfig(CONFIG_SHOW_BAN_IN_WORLD))
+    {
+        bool IsPermanetly = true;
+
+        if (TimeStringToSecs(Duration) > 0)
+            IsPermanetly = false;
+
+        if (IsPermanetly)
+            sWorld->SendWorldText(LANG_BAN_IP_YOUPERMBANNEDMESSAGE_WORLD, Author.c_str(), IP.c_str(), Reason.c_str());
```

Comment:
Cant see this in Miscellaneous/Language.h

Reply:
Okay, i fixed (maybe) :D